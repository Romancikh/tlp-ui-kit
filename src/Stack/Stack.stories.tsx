import type { Meta, StoryObj } from "@storybook/react";

import Stack from "./Stack";

const meta = {
  title: "Stack",
  component: Stack,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
} satisfies Meta<typeof Stack>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: [
      <div key="div1">First Div</div>,
      <div key="div2">Second Div</div>,
      <div key="div3">Third Div</div>,
    ],
  },
};

export const Gap: Story = {
  args: {
    gap: 7,
    children: [
      <div key="div1">First Div</div>,
      <div key="div2">Second Div</div>,
      <div key="div3">Third Div</div>,
    ],
  },
};

export const MarginTop: Story = {
  args: {
    marginTop: 7,
    children: [
      <div key="div1">First Div</div>,
      <div key="div2">Second Div</div>,
      <div key="div3">Third Div</div>,
    ],
  },
};
