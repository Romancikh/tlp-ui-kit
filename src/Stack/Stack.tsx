import MuiStack from "@mui/material/Stack";
import { PropsWithChildren } from "react";

type StackProps = PropsWithChildren & {
  direction?: "column" | "column-reverse" | "row" | "row-reverse";
  marginTop?: number;
  spacing?: number;
};

function Stack({ children, direction, marginTop, spacing }: StackProps) {
  return (
    <MuiStack direction={direction} marginTop={marginTop} spacing={spacing}>
      {children}
    </MuiStack>
  );
}

export default Stack;
