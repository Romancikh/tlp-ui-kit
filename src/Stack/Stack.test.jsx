import { render } from "@testing-library/react";
import Stack from "./Stack";

test("Stack renders correctly", () => {
  const { container } = render(
    <Stack>
      <div key="div1">First Div</div>
      <div key="div2">Second Div</div>
      <div key="div3">Third Div</div>
    </Stack>
  );
  expect(container).toMatchSnapshot();
});

test("Row Stack renders correctly", () => {
  const { container } = render(
    <Stack direction="row">
      <div key="div1">First Div</div>
      <div key="div2">Second Div</div>
      <div key="div3">Third Div</div>
    </Stack>
  );
  expect(container).toMatchSnapshot();
});

test("Spacing Stack renders correctly", () => {
  const { container } = render(
    <Stack spacing={7}>
      <div key="div1">First Div</div>
      <div key="div2">Second Div</div>
      <div key="div3">Third Div</div>
    </Stack>
  );
  expect(container).toMatchSnapshot();
});

test("Margin Stack renders correctly", () => {
  const { container } = render(
    <Stack marginTop={7}>
      <div key="div1">First Div</div>
      <div key="div2">Second Div</div>
      <div key="div3">Third Div</div>
    </Stack>
  );
  expect(container).toMatchSnapshot();
});
