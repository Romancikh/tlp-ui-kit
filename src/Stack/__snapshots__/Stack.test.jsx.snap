// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Margin Stack renders correctly 1`] = `
<div>
  <div
    class="MuiStack-root css-1mvtyz3-MuiStack-root"
  >
    <div>
      First Div
    </div>
    <div>
      Second Div
    </div>
    <div>
      Third Div
    </div>
  </div>
</div>
`;

exports[`Row Stack renders correctly 1`] = `
<div>
  <div
    class="MuiStack-root css-m69qwo-MuiStack-root"
  >
    <div>
      First Div
    </div>
    <div>
      Second Div
    </div>
    <div>
      Third Div
    </div>
  </div>
</div>
`;

exports[`Spacing Stack renders correctly 1`] = `
<div>
  <div
    class="MuiStack-root css-3ouhma-MuiStack-root"
  >
    <div>
      First Div
    </div>
    <div>
      Second Div
    </div>
    <div>
      Third Div
    </div>
  </div>
</div>
`;

exports[`Stack renders correctly 1`] = `
<div>
  <div
    class="MuiStack-root css-nen11g-MuiStack-root"
  >
    <div>
      First Div
    </div>
    <div>
      Second Div
    </div>
    <div>
      Third Div
    </div>
  </div>
</div>
`;
